[WIP] Public beta docs for ephemeral values (#35899)
hashicorp/terraform: website/docs/language/functions/ephemeralasnull.mdx

---
page_title: ephemeralasnull function reference - Functions - Configuration Language  
description: |-  
  The `ephemeralasnull` function accepts an ephemeral value and returns null.  
---

# `ephemeralasnull` function reference

> -> The `ephemeralasnull` function is in public beta and available in Terraform v1.10 and later. The workflows of `ephemeralasnull` are subject to change.

Reviewer: ```suggestion
<Note title="Public beta">

The `ephemeralasnull` function is in public beta and available in Terraform v1.10 and later. The workflows of `ephemeralasnull` are subject to change.

</Note>
```

}
```

For more information on which Terraform blocks can be ephemeral, refer to ephemeral [inputs](/terraform/language/values/variables#exclude-values-from-state), [outputs](/terraform/language/values/outputs#ephemeral-avoid-storing-values-in-state-or-plan-files), and [resources](/terraform/language/resources/ephemeral).

## Example use case

The following example shows how you can use the  `ephemeralasnull` function to intake a map that contains some ephemeral values, and output the values that are not ephemeral.

```hcl
variable "session_token" {
  type      = string
  default   = "test"
  ephemeral = true
}

variable "configuration" {
  type = map(string)
  default = {
    "env" = "development"
  }
}

# This is a contrived example, but this pattern works with any object that is a mix of ephemeral and non-ephemeral values. 
locals {
  configuration_with_token = merge(
    var.configuration,
    { "session_token" = var.session_token }
  )
  ephemeral = true
}

output "configuration_settings" {
# Using ephemeralasnull enables you to output the non-ephemeral values.
  value       = ephemeralasnull(local.things_with_token)
  description = "Environment setting."
}
```

> Applying the above configuration returns the `ephemeral` values in `locals.configuration_settings` as `null`.
> 
> ```hcl
> Outputs:
> 
> configuration_settings = {
>   "env" = "development"
>   "session_token" = tostring(null)
> }
> ```
> Allowing you to output the non-ephemeral value `env`.

Reviewer: ````suggestion
When you apply the above configuration, Terraform returns the `ephemeral` values in `locals.configuration_settings` as `null`. This lets you output the non-ephemeral value `env`.

```hcl
Outputs:

configuration_settings = {
  "env" = "development"
  "session_token" = tostring(null)
}
```
````
